src/utils/context.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState } from 'react';
+import usePersistStorage from 'react-native-use-persist-storage';
+
+export type TPersistContext<T> = {
+  data: T;
+  setData: React.Dispatch<React.SetStateAction<T>>;
+  restored: boolean;
+};
+
+export const createPersistContext = <T extends {}>({
+  defaultData,
+  storageKey,
+  version = 0,
+}: {
+  defaultData: T;
+  storageKey: string;
+  version?: number;
+}) => {
+  const createDefaultData = () => defaultData;
+
+  const Context = React.createContext<TPersistContext<T>>({
+    data: createDefaultData(),
+    setData: () => {
+      return;
+    },
+    restored: false,
+  });
+
+  const Provider: React.FC<{
+    persist?: boolean;
+  }> = props => {
+    const [data, setData, restored] = usePersistStorage<T>(
+      storageKey,
+      createDefaultData,
+      {
+        persist: props.persist,
+        version,
+      },
+    );
+
+    return (
+      <Context.Provider value={{ data, setData, restored }}>
+        {props.children}
+      </Context.Provider>
+    );
+  };
+
+  const useData = () => {
+    const context = React.useContext(Context);
+    if (!context) {
+      throw new Error(
+        `Context error: context [${storageKey}] must be used within a Provider`,
+      );
+    }
+    const { data, setData, restored } = context;
```

Comment:
If our context format is `[data, setData, restored]`.
I think we can directly return the whole context, cause it is clear.

And our use case may be like

 
`screen.tsx`
```javascript
import { useUser } from '@/contexts/userContext';
// ...
const [user, setUser, restored] = useUser(); 
// ...
```


also, if some body want to custom his hook, he can

`user.ts`
```
import { UserContext } from '@/contexts/userContext';
// ...
const [user, setUser, restored] = useContext(UserContext); 
const setUserById = () => ({ ... });
return {
  user,
  setUser,
  setUserById,
  ...
}
// ...
```